### ISSUE TITLE
Add multiple items to cart `addCartItems`

### ISSUE THREAD

**Issue by thetre97** [wp-graphql/wp-graphql-woocommerce#339]
**Is your feature request related to a problem? Please describe.**
I have some grouped products, and a customer can add multiple products to a cart at once.
I currently have to loop through these and perform 3-4 mutations to add all products to cart, instead of being to specify multiple products at once.

**Describe the solution you'd like**
Either the current `addToCart` could support multiple items (like `input: [AddToCartInput]`) or another mutation like `addItemsToCart` which supports multiple items.

I'd be happy to try and implement this if it is just a case of looping over the items and performing the same actions as the `addToCart` mutation.

**Comment by MobyDigg** [comment]
Hi @thetre97, I have a similar problem. How did you solve this? With a promise? Would be glad to exchange some ideas about this topic… thanks.

**Comment by thetre97** [comment]
@MobyDigg Yes, I just did a map of the products I needed to add, and did `Promise.all` (this is with Vue Apollo):
```js
const addToCartMutations = productPayloads.map(input => this.$apollo.mutate({
	mutation: AddToCartMutation,
	variables: { input },
	update: (store, { data: { addToCart } }) => {
		store.writeQuery({ query: CartSimpleQuery, data: addToCart })
		// Need to manually update product inventory
		store.writeQuery({ query: ProductQuery, variables: { id: this.$context.id }, data: { product: addToCart.cartItem.product } })
	}
}))

const results = await Promise.all(addToCartMutations)
```

**Comment by pandabrand** [comment]
@thetre97 , @MobyDigg have either of you run into issues sending multiple addToCart mutations? Recently mine have started to time out.

**Comment by thetre97** [comment]
I haven't, no - but I only ever had a maximum of 2-3 products to add... 

Are you adding more than that?

**Comment by MobyDigg** [comment]
> @thetre97 , @MobyDigg have either of you run into issues sending multiple addToCart mutations? Recently mine have started to time out.

No not really, are you using the WooCommerce session token in your headers?

**Comment by pandabrand** [comment]
@MobyDigg I am using the session token in my http request headers.

**Comment by thetre97** [comment]
Closed in #439
